File: backend/core/handler/outstanding_moeny/delete.py
from core.data.response import ResponseData
from core.repository.psql.outstanding_moeny.delete import delete_list_psql, delete_item_psql
from core.data.user import UserData


def handler_delete_list(user_data: UserData, id: str) -> ResponseData:
    try:
        response_delete = delete_list_psql(user_data, id)
        if not response_delete['is_valid']:
            return ResponseData(
                is_valid=response_delete['is_valid'],
                status=response_delete['status'],
                data=response_delete['data'],
                status_code=response_delete['status_code'],
                additional=response_delete['additional']
            )

        return ResponseData(
            is_valid=response_delete['is_valid'],
            status=response_delete['status'],
            data=response_delete['data'],
            status_code=response_delete['status_code'],
            additional=response_delete['additional']
        )
    except Exception as e:
        return ResponseData(
            is_valid=False,
            status="ERROR",
            data=str(e),
            status_code=500,
            additional=None
        )


def handler_delete_item(user_data: UserData, id: str) -> ResponseData:
    try:
        response_delete = delete_item_psql(user_data, id)
        if not response_delete['is_valid']:
            return ResponseData(
                is_valid=response_delete['is_valid'],
                status=response_delete['status'],
                data=response_delete['data'],
                status_code=response_delete['status_code'],
                additional=response_delete['additional']
            )

        return ResponseData(
            is_valid=response_delete['is_valid'],
            status=response_delete['status'],
            data=response_delete['data'],
            status_code=response_delete['status_code'],
            additional=response_delete['additional']
        )
    except Exception as e:
        return ResponseData(
            is_valid=False,
            status="ERROR",
            data=str(e),
            status_code=500,
            additional=None
        )

File: frontend/src/api/outstandingMoney/delete.ts
import { apiPost } from "@/api/common/post";
import type { ResponseData, Error } from "@/types/global";
import type {
  ApiDeleteList,
  ApiDeleteItem,
} from "@/types/api/outstandingMoney/types";

export async function outStandingMoneyDeleteList(
  id: string,
): Promise<ResponseData> {
  const urlPath = `/outstanding_money/delete_list/${id}`;
  const response = await apiPost(urlPath, {}, "DELETE", 0, {
    Authorization: true,
    UserData: true,
  });

  if (
    !response ||
    response.status !== "SUCCESS" ||
    response.status_code >= 400
  ) {
    console.error("API response does not return delete list!");
    return {
      isValid: false,
      data: response.data as Error,
      additional: response.additional,
    };
  }

  return {
    isValid: true,
    data: response.data as ApiDeleteList,
    additional: response.additional,
  };
}

export async function outStandingMoneyDeleteItem(
  id: string,
): Promise<ResponseData> {
  const urlPath = `/outstanding_money/delete_item/${id}`;
  const response = await apiPost(urlPath, {}, "DELETE", 0, {
    Authorization: true,
    UserData: true,
  });

  if (
    !response ||
    response.status !== "SUCCESS" ||
    response.status_code >= 400
  ) {
    console.error("API response does not return delete item!");
    return {
      isValid: false,
      data: response.data as Error,
      additional: response.additional,
    };
  }

  return {
    isValid: true,
    data: response.data as ApiDeleteItem,
    additional: response.additional,
  };
}
